tools/mkdfs/mkdfs.c

```diff
@@ -1,20 +1,30 @@
 #include <stdio.h>
-#include <stdlib.h>
-#include <dirent.h>
+#include <malloc.h>
 #include <sys/stat.h>
 #include <string.h>
 #include <errno.h>
 #include <stdint.h>
+
 #include <sys/types.h>
+#ifndef _MSC_VER
+#include <dirent.h>
 #include <sys/param.h>
+#else
+#include "dirent.h"
+#endif // !_MSC_VER
+
 #include "dragonfs.h"
 #include "dfsinternal.h"
 
+#ifndef _MSC_VER
 #if BYTE_ORDER == BIG_ENDIAN
 #define SWAPLONG(i) (i)
 #else
 #define SWAPLONG(i) (((uint32_t)(i & 0xFF000000) >> 24) | ((uint32_t)(i & 0x00FF0000) >>  8) | ((uint32_t)(i & 0x0000FF00) <<  8) | ((uint32_t)(i & 0x000000FF) << 24))
 #endif
+#else
```

Comment:
Can't see the difference here, am I blind :)

Reply:
MSVC does not recognise the flag BIG_ENDIAN and there is no simple way to detect otherwise. Since all Windows systems are little endian, this was the easiest option.

Reply:
for reference: https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html

https://stackoverflow.com/questions/8978935/detecting-endianness

![image](https://user-images.githubusercontent.com/11439699/84516196-5999ae00-acc5-11ea-97ae-aa1bb3b16e12.png)

Reply:
This makes sense now :)